packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -404,63 +493,261 @@ const handler: AdaptingProxyHandler<ProxyTarget, EditableTree> = {
+        fieldSchema: FieldSchema,
+        cursor: ITreeSubscriptionCursor,
+        primaryType?: TreeSchemaIdentifier,
+    ) {
+        assert(cursor.mode === CursorLocationType.Fields, "must be in fields mode");
+        super(context, cursor);
+        this.fieldKey = cursor.getFieldKey();
+        this.primaryType = primaryType;
+        this.fieldSchema = fieldSchema;
+    }
+
+    buildAnchorFromCursor(cursor: ITreeSubscriptionCursor): FieldAnchor {
+        return cursor.buildFieldAnchor();
+    }
+
+    tryMoveCursorToAnchor(
+        anchor: FieldAnchor,
+        cursor: ITreeSubscriptionCursor,
+    ): TreeNavigationResult {
+        return this.context.forest.tryMoveCursorToField(anchor, cursor);
+    }
+
+    forgetAnchor(anchor: FieldAnchor): void {
+        if (anchor.parent === undefined) return;
+        this.context.forest.anchors.forget(anchor.parent);
+    }
+
+    readonly [index: number]: UnwrappedEditableTree;
+
+    public get length(): number {
+        return this.cursor.getFieldLength();
+    }
+
+    /**
+     * Returns a node (unwrapped by default, see {@link UnwrappedEditableTree}) by its index.
+     */
+    public proxifyNode(index: number, unwrap = true): UnwrappedEditableTree {
+        this.cursor.enterNode(index);
+        const target = new NodeProxyTarget(this.context, this.cursor);
+        this.cursor.exitNode();
+        return inProxyOrUnwrap(target, unwrap);
+    }
+
+    /**
+     * Gets a node by its index without unwrapping.
+     */
+    public getWithoutUnwrapping(index: number): EditableTree {
+        assert(
+            keyIsValidIndex(index, this.length),
+            "A child node must exist at index to get it without unwrapping.",
+        );
+        return this.proxifyNode(index, false) as EditableTree;
+    }
+
+    /**
+     * Gets array of unwrapped nodes.
+     */
+    private get asArray(): UnwrappedEditableTree[] {
+        return mapCursorField(this.cursor, (c) =>
+            inProxyOrUnwrap(new NodeProxyTarget(this.context, c), true),
+        );
+    }
+
+    *[Symbol.iterator](): IterableIterator<UnwrappedEditableTree> {
+        for (const node of this.asArray) {
+            yield node;
+        }
+    }
+}
+
+/**
+ * Returns a Proxy handler, which together with a {@link FieldProxyTarget} implements a basic read/write access to
+ * the sequence fields by means of the cursors.
+ */
+const fieldProxyHandler: AdaptingProxyHandler<FieldProxyTarget, EditableField> = {
+    get: (target: FieldProxyTarget, key: string | symbol, receiver: object): unknown => {
+        if (typeof key === "string") {
+            if (key in { length: true, fieldKey: true, fieldSchema: true, primaryType: true }) {
+                return Reflect.get(target, key);
+            } else if (keyIsValidIndex(key, target.length)) {
+                return target.proxifyNode(Number(key));
+            }
+            const reflected = Reflect.get(target, key);
+            if (typeof reflected === "function") {
```

Comment:
this will expose access to private functions, and stuff from the prototype like toString. It would be nice to be more clear here. Maybe document why this is ok and what we want to do, or handle the functions we explicitly want to expose?